### PR TITLE
TASK-53132: Enable creating and posting announcement challenge with emoji .

### PR DESCRIPTION
**Pull request by ArfaouiMohamedAmine**
when an emoji is added to the description to announce or to create a challenge, an `sql` error has been endorsed, it turns out that the description (named `COMMENT` ,`EVENT` ,`DESCRIPTION`, `ACTION_TITLE ` in the `GAMIFICATION_RULE`,`GAMIFICATION_ACTIONS_HISTORY` tables) is collated as `utf8`, in order to resolve this issue ; a type modification of utf8mb4 COLLATE `utf8mb4_unicode_ci` and a type change to `varchar`(2000) to support emojis has been added to the `changelog` .

### CODE REVIEW

File: services/src/main/resources/db/changelog/gamification.db.changelog-1.0.0.xml [hunk 1]
```diff
@@ -419,9 +419,29 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
         <addAutoIncrement columnDataType="BIGINT" columnName="CHALLENGE_MANAGER_ID" incrementBy="1" startWith="1"
                           tableName="CHALLENGE_MANAGER_RULE"/>
     </changeSet>
-    <changeSet id="1.0.0-37" author="exo-gamification" dbms="mysql">
+    <changeSet id="1.0.0-38" author="exo-gamification" dbms="mysql">
```
Comment (GouadriaHanen) [inline services/src/main/resources/db/changelog/gamification.db.changelog-1.0.0.xml]: changeSet with id 37 has been executed so you have to add a new changeset without removing the last one

File: services/src/main/resources/db/changelog/gamification.db.changelog-1.0.0.xml [hunk 2]
```diff
@@ -419,9 +419,29 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
...
         <modifyDataType   columnName="COMMENT"
-                          newDataType="NVARCHAR(2000) SET utf8mb4 COLLATE utf8mb4_unicode_ci"
+                          newDataType="VARCHAR(2000) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci"
+                          tableName="GAMIFICATION_ACTIONS_HISTORY"/>
+    </changeSet>
+    <changeSet id="1.0.0-39" author="exo-gamification" dbms="mysql">
```
Comment (GouadriaHanen) [inline services/src/main/resources/db/changelog/gamification.db.changelog-1.0.0.xml]: you can add all the modifications in the same changeSet

File: services/src/main/resources/db/changelog/gamification.db.changelog-1.0.0.xml [hunk 3]
```diff
@@ -419,9 +419,29 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
...
+        <modifyDataType   columnName="EVENT"
+                          newDataType="VARCHAR(2000) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci"
+                          tableName="GAMIFICATION_RULE"/>
+    </changeSet>
+    <changeSet id="1.0.0-40" author="exo-gamification" dbms="mysql">
```
Comment (GouadriaHanen) [inline services/src/main/resources/db/changelog/gamification.db.changelog-1.0.0.xml]: same here

File: services/src/main/resources/db/changelog/gamification.db.changelog-1.0.0.xml [hunk 4]
```diff
@@ -419,9 +419,29 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
...
+        <modifyDataType   columnName="DESCRIPTION"
+                          newDataType="VARCHAR(2000) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci"
+                          tableName="GAMIFICATION_RULE"/>
+    </changeSet>
+    <changeSet id="1.0.0-41" author="exo-gamification" dbms="mysql">
```
Comment (GouadriaHanen) [inline services/src/main/resources/db/changelog/gamification.db.changelog-1.0.0.xml]: same here

File: services/src/main/resources/db/changelog/gamification.db.changelog-1.0.0.xml [hunk 5]
```diff
@@ -419,9 +419,29 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
...
+        <modifyDataType   columnName="TITLE"
+                          newDataType="VARCHAR(2000) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci"
+                          tableName="GAMIFICATION_RULE"/>
+    </changeSet>
+    <changeSet id="1.0.0-42" author="exo-gamification" dbms="mysql">
```
Comment (GouadriaHanen) [inline services/src/main/resources/db/changelog/gamification.db.changelog-1.0.0.xml]: same here

**Review by GouadriaHanen — CHANGES_REQUESTED** [review]

File: services/src/main/resources/db/changelog/gamification.db.changelog-1.0.0.xml (hunk 1 above)
Comment (GouadriaHanen) [reply services/src/main/resources/db/changelog/gamification.db.changelog-1.0.0.xml]: as verified in the acceptance DB changelog and acceptance log : changeSet with 37 had not been executed so you can use 37 as id

File: services/src/main/resources/db/changelog/gamification.db.changelog-1.0.0.xml (hunk 2 above)
Comment (ArfaouiMohamedAmine) [reply services/src/main/resources/db/changelog/gamification.db.changelog-1.0.0.xml]: done !

File: services/src/main/resources/db/changelog/gamification.db.changelog-1.0.0.xml (hunk 1 above)
Comment (ArfaouiMohamedAmine) [reply services/src/main/resources/db/changelog/gamification.db.changelog-1.0.0.xml]: done !

**Review by ahamdi — APPROVED** [review]

**Pull request merged**